src/__tests__/RESTDataSource.test.ts

```diff
@@ -780,6 +780,44 @@ describe('RESTDataSource', () => {
         ]);
       });
 
+      it('allows specifying a custom cache key via cacheKey used for deduplication', async () => {
+        const dataSource = new (class extends RESTDataSource {
+          override baseURL = 'https://api.example.com';
+
+          getFoo(id: number) {
+            return this.get(`foo/${id}`, {
+              cacheKey: 'constant',
+            });
+          }
+        })();
+
+        nock(apiUrl).get('/foo/1').reply(200);
+
+        await Promise.all([dataSource.getFoo(1), dataSource.getFoo(1)]);
```

Comment:
I think as-is this test would still pass without the `cacheKey` override. What if we did this to show the cache key we provided is actually being used (instead of the url)?

```suggestion
        await Promise.all([dataSource.getFoo(1), dataSource.getFoo(2)]);
```

Reply:
very good point, took your suggestions.